src/revocation.rs

```diff
@@ -0,0 +1,477 @@
+/// revocation list, e.g. in a [RevocationList2020].
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(try_from = "String")]
+#[serde(into = "String")]
+pub struct RevocationListIndex(usize);
+
+/// Verifiable Credential of type RevocationList2020Credential.
+/// <https://w3c-ccg.github.io/vc-status-rl-2020/#revocationlist2020credential>
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct RevocationList2020Credential {
+    pub id: URI,
+    pub issuer: Issuer,
+    pub credential_subject: RevocationList2020Subject,
+    #[serde(flatten)]
+    pub more_properties: Value,
+}
+
+/// [Credential subject](https://www.w3.org/TR/vc-data-model/#credential-subject) of a [RevocationList2020Credential]
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(tag = "type")]
+pub enum RevocationList2020Subject {
+    RevocationList2020(RevocationList2020),
+}
+
+/// Credential subject of type RevocationList2020, expected to be used in a Verifiable Credential of type [RevocationList2020Credential]
+/// <https://w3c-ccg.github.io/vc-status-rl-2020/#revocationlist2020credential>
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct RevocationList2020 {
+    pub encoded_list: EncodedList,
+    #[serde(flatten)]
+    pub more_properties: Value,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct EncodedList(pub String);
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(tag = "type")]
+pub enum RevocationSubject {
+    RevocationList2020(RevocationList2020),
+}
+
+/// A decoded [revocation list][EncodedList].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct List(pub Vec<u8>);
+
+impl TryFrom<String> for RevocationListIndex {
+    type Error = std::num::ParseIntError;
+    fn try_from(string: String) -> Result<Self, Self::Error> {
+        Ok(Self(string.parse()?))
+    }
+}
+
+impl From<RevocationListIndex> for String {
+    fn from(idx: RevocationListIndex) -> String {
+        idx.0.to_string()
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum SetStatusError {
+    #[error("Encode list: {0}")]
+    Encode(#[from] EncodeListError),
+    #[error("Decode list: {0}")]
+    Decode(#[from] DecodeListError),
+    #[error("Out of bounds: index {0} but length is {1}")]
+    OutOfBounds(usize, usize),
+    #[error("Revocation list too large for BitVec: {0}")]
+    ListTooLarge(usize),
+    #[error("Revocation list is too small: {0}. Minimum: {1}")]
+    ListTooSmall(usize, usize),
+}
+
+impl RevocationList2020 {
+    /// Set the revocation status for a given index in the list.
+    pub fn set_status(&mut self, index: usize, revoked: bool) -> Result<(), SetStatusError> {
+        let mut list = List::try_from(&self.encoded_list)?;
+        let byte_len = list.0.len();
+        if byte_len < MIN_BITSTRING_SIZE_BYTES {
+            return Err(SetStatusError::ListTooSmall(
+                byte_len * 8,
```

Comment:
Bit confusing to have the check use `byte_len` and the error to refer to `byte_len * 8`

Reply:
Addressing in f01f86bc491ce0d164160dea739a716aec00f8da